src/plugins/vis_augmenter/public/view_events_flyout/actions/open_events_flyout.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React from 'react';
+import { toMountPoint } from '../../../../opensearch_dashboards_react/public';
+import { ViewEventsFlyout } from '../components';
+import { VIEW_EVENTS_FLYOUT_STATE, setFlyoutState } from '../flyout_state';
+import { getCore } from '../../services';
+
+interface Props {
+  savedObjectId: string;
+}
+
+export async function openViewEventsFlyout(props: Props) {
+  setFlyoutState(VIEW_EVENTS_FLYOUT_STATE.OPEN);
+  const flyoutSession = getCore().overlays.openFlyout(
```

Comment:
Not related to this PR. But I wonder if an enhancement through this repo would be to have setters and then access it through a services class instead of passing down services.

Reply:
Fair enough - overall, I've seen the strategy I used more commonly across the other internal plugins. If there's an updated guidance for that, I think it's worth having a meta issue to standardize across all plugins.

It was pretty overwhelming to see all of the different ways services were being accessed across all of the different plugins

Reply:
Yeah setters and getters is a common pattern in core but thats a code style preference more than it is a functional requirement. This works too. If we see it pop up in more places we can think about how we want to consolidate them.